factom_sdk/client/chains_client.py

```diff
@@ -62,60 +61,109 @@ def create(self, content: str, external_ids: list = None, signer_private_key: st
         Returns:
             Chain created info object.
         """
-        if callback_stages is None:
-            callback_stages = []
-        if external_ids is None:
-            external_ids = []
-        if self.automatic_signing:
-            if not isinstance(external_ids, list):
-                raise Exception("external_ids must be an array.")
-            if not signer_private_key:
-                raise Exception("signer_private_key is required.")
+        # if external_ids is None:
+        #     external_ids = []
+        
+        external_ids            = kwargs.get('external_ids', [])
+        signer_private_key      = kwargs.get('signer_private_key', None)
+        signer_chain_id         = kwargs.get('signer_chain_id', None)
+
+        
+
+        # if callback_stages is None:
+        #     callback_stages = []    
+        callback_stages         = kwargs.get('callback_stages', [])
+        callback_url            = kwargs.get('callback_url', "")
+
+        automatic_signing       = kwargs.get('automatic_signing', self.automatic_signing)
+
+        # User should be able to pass any sequence of elements
+        if not hasattr(external_ids, '__iter__'):
+            raise Exception("external_ids must be iterable")
+
+        if automatic_signing:
+            if not (signer_private_key is not None and signer_chain_id is not None):
+                raise Exception("Must pass both private key and identity if chain is to be signed")
+            # if not isinstance(external_ids, list):
+            #     raise Exception("external_ids must be an array.")
+            # if not signer_private_key:
+            #     raise Exception("signer_private_key is required.")
             if not KeyCommon.validate_checksum(signer_private_key):
                 raise Exception("signer_private_key is invalid.")
-            if not signer_chain_id:
-                raise Exception("signer_chain_id is required.")
+            # if not signer_chain_id:
+            #     raise Exception("signer_chain_id is required.")
         else:
-            if not external_ids:
-                raise Exception("at least 1 external_id is required.")
-            if not isinstance(external_ids, list):
-                raise Exception("external_ids must be an array.")
-            if signer_private_key and not signer_chain_id:
-                raise Exception("signer_chain_id is required when passing a signer_private_key.")
-            if signer_private_key and not KeyCommon.validate_checksum(signer_private_key):
-                raise Exception("signer_private_key is invalid.")
-            if signer_chain_id and not signer_private_key:
-                raise Exception("signer_private_key is required when passing a signer_chain_id.")
-        if not content:
-            raise Exception("content is required.")

```

Comment:
Why was the None-check on `content` removed?

Reply:
My bad, that was an error on my part. I realize now that the None check for content should remain.
My rationale was that since it's a compulsory function arg, there would always be some content passed by the user. But in the case the user passes in "None" the None-check would handle that.